### DISCUSSION TITLE
Can we use the supabase client together with SWR or React-query?

### DISCUSSION THREAD

**Discussion by cloudorbush** [supabase/supabase#936] [Questions] [answered]
Any insight would be greatly appreciated.

Thanks!

**Comment by cloudorbush** [comment, marked as answer]
Found this on [Twitter ](https://twitter.com/eunjae_lee/status/1369788285317419009)in case anybody has the same issue.

![image](https://user-images.githubusercontent.com/57633368/112360524-cad0ad80-8ca8-11eb-8afc-bbdb87a03b0c.png)

**Comment by psteinroe** [comment]
If you plan to use swr with supabase, be sure to check out [Supabase Cache Helpers](https://github.com/psteinroe/supabase-cache-helpers). It integrates supabase-js with swr and comes with mutation, subscription and convenience hooks for pagination as well as storage support. All logic is split up in reusable packages, hence adding support for `react-query` should be quite straight forward too.

```tsx
   // Define the query. Check out the other query hooks for pagination etc.
    const { data, count } = useQuery(
    client
        .from("contact")
        .select("id,username,ticket_number", { count: "exact" })
        .eq("username", "myusername"),
    "multiple",
    {
        revalidateOnFocus: false,
        revalidateOnReconnect: false,
    }
    );

    // Define the mutation. Update, Upsert, and Delete are also supported.
    const [insert] = useInsertMutation(client.from("contact"), "single");

    // Subscriptions are also supported.
    const { status } = useSubscription(
        client.channel('random'),
        {
            event: "*",
            table: "contact",
            schema: "public",
        },
        ["id"],
        { callback: (payload) => console.log(payload) }
    );

```
